File: algorithms/2_medium/N/00435_non_overlapping_intervals/two_pointers_greedy.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](const auto &a, const auto &b) {
            if (a[0] == b[0]) {
                return a[1] < b[1];
            }

            return a[0] < b[0];
        });

        size_t left{0};
        const size_t len{intervals.size()};
        int count {0};

        for (size_t right{1}; right < len; ++right) {
            if (intervals[left][1] <= intervals[right][0]) {
                left = right;
            } else {
                if (intervals[right][1] < intervals[left][1]) {
                    left = right;
                }

                count += 1;
            }
        }

        return count;
    }
};

File: algorithms/2_medium/N/00435_non_overlapping_intervals/two_pointers_greedy.rs
impl Solution {
    pub fn erase_overlap_intervals(mut intervals: Vec<Vec<i32>>) -> i32 {
        intervals.sort_by(|a, b| {
            if (a[0] == b[0]) {
                a[1].cmp(&b[1])
            } else {
                a[0].cmp(&b[0])
            }
        });

        let mut left = 0_usize;
        let len = intervals.len();
        let mut count = 0_i32;

        for right in 1..len {
            if intervals[left][1] <= intervals[right][0] {
                left = right;
            } else {
                if intervals[right][1] < intervals[left][1] {
                    left = right;
                }

                count += 1;
            }
        }

        count
    }
}
